utils/transformedPayload.js

```diff
@@ -0,0 +1,24 @@
+const transformPayload = (payload, referenceData) => {
+    // let transformedPayload = JSON.parse(JSON.stringify(payload));
+    let transformedPayload = payload;
+    const referenceKeys = Object.keys(referenceData);
+
+    // Transofrmation logic   
+
+    // return if key != value
+    if (typeof transformedPayload.value !== 'string') {
+        transformedPayload.value.forEach((valueObj) => { valueObj = transformPayload(valueObj, referenceData) })
+    } else {
+        // check if value of 'value' key is present in referenceKeys
+        referenceKeys.forEach(refKey => {
```

Comment:
your looping through each property in `referenceData` for every string value in `payload`. If the `referenceData` contains 100's of properties but your payload may not necessarily contain all of these properties is it worth looping through all ? Can this be made more efficient ?